pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -485,6 +469,11 @@ func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client,
 	if !p.isGroup {
 		podSets = p.PodSets()
 	} else {
+		// If the whole group has finished, don't create a workload
+		if _, finished := p.Finished(); finished {
```

Comment:
I don't think this behavior is special for pod groups.
If any job is finished, we shouldn't be creating a Workload.

I think this should have been covered here https://github.com/kubernetes-sigs/kueue/blob/881288427cdbf94b24c5b56654287d02a3272ae1/pkg/controller/jobframework/reconciler.go#L453

Can you debug why it's not?

Reply:
If the job succeeds, it's considered suspended, so the reconciler won't go into that branch:
https://github.com/kubernetes-sigs/kueue/blob/881288427cdbf94b24c5b56654287d02a3272ae1/pkg/controller/jobframework/reconciler.go#L443

Let me see what we can do about that.

Reply:
Uhm... I guess most jobs are not "suspended" when they are finished.

We might need to modify the logic in the generic reconciler a little to always finalize jobs and return when the Workload is finished.

Also, no need to return an error in this case, even if the error is not retriable.
Error means that something went wrong. Nothing went wrong here, we just don't want to do anything, and that can be done with a simple `return ctrl.Result{}, nil`